code/PostProcessing/JoinVerticesProcess.cpp

```diff
@@ -100,6 +100,48 @@ void JoinVerticesProcess::Execute( aiScene* pScene) {
 
 namespace {
 
+struct CompareVerticesAlmostEqual {
+    bool operator () (const Vertex & a, const Vertex & b) const {
+        static const float epsilon = 1e-5f;
+        static const float squareEpsilon = epsilon * epsilon;
+
+        if ((a.position - b.position).SquareLength() > squareEpsilon) {
+            return (a.position < b.position);
+        }
+
+        if ((a.normal - b.normal).SquareLength() > squareEpsilon) {
+            return (a.normal < b.normal);
+        }
+
+        // We just test the other attributes even if they're not present in the mesh.
+        // In this case they're initialized to 0 so the comparison succeeds.
+        // By this method the non-present attributes are effectively ignored in the comparison.
+
+        if ((a.tangent - b.tangent).SquareLength() > squareEpsilon) {
```

Comment:
I added back the checks for `tangent` and `bitangent` since the pre 12f33099964c50877ff9bc29957212e020deb6be logic checked for them. Per the comment (also restored from before), these will have no effect if they are not set as there will be no difference between them.